metrics/config_observability.go

```diff
@@ -83,6 +87,14 @@ func NewObservabilityConfigFromConfigMap(configMap *corev1.ConfigMap) (*Observab
 		return nil, err
 	}
 
+	if raw, ok := configMap.Data["logging.enable-request-log"]; ok {
+		if strings.EqualFold(raw, "true") && oc.RequestLogTemplate != "" {
+			oc.EnableRequestLog = true
+		}
+	} else if oc.RequestLogTemplate != "" {
+		oc.EnableRequestLog = true
+	}
```

Comment:
What's our plan to distinguish from the default value here? Eventually, this will always be true because we'll set an actual example template that's not empty.

One possibility is to leave this in for 0.17 and then in 0.18, remove these lines and only work off of the switch, calling out in 0.17s release-notes that the switch **must** be set to true for request logging to work come 0.18.

If we want to do it that way, let's add a comment here stating we're going to remove this implicit defaulting after 0.17 cuts (and that we'll provide a proper default for the template once 0.17 cuts too).

Reply:
@markusthoemmes this follows the suggestion stated above:

> If the template is set I guess we should default to true, otherwise to false to be able to upgrade safely?

> What's our plan to distinguish from the default value here? Eventually, this will always be true because we'll set an actual example template that's not empty.

The default value is `false` and we start with that assumption of not having logging enabled. So the user either sets the flag to true and then we check if a template exists or if he sets the template we turn the flag to true.

Reply:
@markusthoemmes alternatively we could avoid changing the default `true` if a template is empty and emit a warning saying that this requires a template to be set or we could use a default template if none is used so logging can work by default. 
This way downstream, if people have a template, things will not break but also we will not need to remove anything in the future. This way if user wants to disable logging needs to set the flag to `false` in any version (current or future). It is the opposite of what we have right now. WDYTH?

Reply:
Added a comment btw.

Reply:
Can we keep the template empty by default and add a template example in the comments in `_example` instead? If we go this way, we can enable-request-log defaults to false if template is empty and defaults to true if template is not empty.

Reply:
@yanweiguo for step one, yes. Eventually we don't want that though. The whole point of this is to be able to have _example == default so we'll have to move to a mode where that works eventually.